---
title: "Proyecto 1: el microbioma del pulque"
author: "Patricia Bustos, Montserrat García, Celso Cortés, Antonio Ovalle, Tobias Portillo, Eduardo Juscamayta, Armando Ocampo"
institute: "CDSD 2022"
date: "`r Sys.Date()`"
output:
  xaringan::moon_reader:
    css: [xaringan-themer.css]
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      

---
```{r xaringan-themer, include=FALSE, warning=FALSE}
library(xaringanthemer)
style_mono_accent(
  base_color = "#1c5253",
  header_font_google = google_font("Josefin Sans"),
  text_font_google   = google_font("Montserrat", "300", "300i"),
  code_font_google   = google_font("Fira Mono")
)
```

```{r xaringanExtra-clipboard, echo=FALSE}
# No quitar! Es para que aparezca la opción copy code en los R chunks
xaringanExtra::use_clipboard()
```


# Flujo de trabajo

a) MaxBin

b) MetaBat

c) DASTool

d) Extracción de Bins

e) Taxonomía

f) Inferencia metabólica

g) Anotación

---

# MaxBin

Generación de enlaces simbólicos

```{r MAX, eval=FALSE}
 mkdir -p 01.Metabat/{data,results}
 cd 01.Metabat/
 ln -s /home/diana/samples/pulque/data/pulque.fasta data/  
 ln -s /home/diana/samples/pulque/data/pulque_sorted.bam  data/ 
```

Activación de ambiente

```{r max1, eval=FALSE}
conda activate maxbin_env
```

Cálculo de bins

```{r max2, eval=FALSE}
run_MaxBin.pl -contig data/pulque.fasta -out results/maxbin -abund data/pulque-depth.txt -max_iteration 2
```

---

# MetaBat

```{r meta, eval=FALSE}
mkdir -p 02.Metabat/{data,results}
cd 02.Metabat/
ln -s /home/diana/samples/pulque/data/pulque.fasta data/  
ln -s /home/diana/samples/pulque/data/pulque_sorted.bam  data/ 
```

Activamos el ambiente
```{r meta2, eval=FALSE}
conda activate metabat_env
```

Ensamble y archivos de profundidad

```{r meta3, eval=FALSE}
jgi_summarize_bam_contig_depths --outputDepth data/pulque-depth.txt data/pulque_sorted.bam

metabat -i data/pulque.fasta -a data/pulque-depth.txt -o results/bins -t 4 --minCVSum 0 --saveCls -d -v --minCV 0.1 -m 2000
```

---
# DASTool

Refinamiento


Preparar datos de entrada

```{r das, eval=FALSE}

Fasta_to_Scaffolds2Bin.sh -i /home/user01/01.Metabat -e fa > htn.scaffolds2bin.tsv
```


```{r das2, eval=FALSE}
PATH=/home/programs:$PATH

cd /home/programs/DAS_Tool-1.1.2

R CMD INSTALL package/DASTool_1.1.2.tar.gz

/home/programs/DAS_Tool-1.1.2/DAS_Tool -i pulque_maxbin.contigs2bin.tsv,pulque_metabat.scaffolds2bin.tsv -l maxbin,metabat -c data/pulque.fasta -o results/pulque_bins --debug -t 4  --search_engine diamond --write_bins 1 
```


---

# Extracción de Bins

```{r che, eval=FALSE}
mkdir 03.CheckM

cd 03.CheckM/

```

Activamos el ambiente

```{r che2, eval=FALSE}
conda activate checkm_env
```

```{r che3, eval=FALSE}
checkm  lineage_wf -t 4 -x fa /home/user01/04.DAS_tool/results/pulque_bins_DASTool_bins DAStools-log_htn  -f CheckM-DAS_Tool_bins.txt

sed -e '1,3d' CheckM-DAS_Tool_bins.txt | sed -e '37d' >CheckM-DAS_Tool_bins_mod.txt
```

---

# Extracción de Bins

```{r studio, eval=FALSE}
library(tidyverse)


checkm<-read.table("CheckM-DAS_Tool_bins_mod.txt", sep = "", header = F, na.strings ="", stringsAsFactors= F)


colnames(checkm)<-c("Bin_Id", "Marker", "lineage", "Number_of_genomes", 
                         "Number_of_markers", "Number_of_marker_sets", 
                         "0", "1", "2", "3", "4", "5", "Completeness", 
                         "Contamination", "Strain_heterogeneity")  

good_bins<-checkm %>%
  select(Bin_Id, Marker, Completeness, Contamination) %>%
  filter(Completeness >= 50.00) %>%
  filter(Contamination <= 10.00) 





```

---

# Extracción de Bins

```{r studio2, eval=FALSE}
medium_bins<-checkm %>%
  select(Bin_Id, Marker, Completeness, Contamination) %>%
  filter(Completeness >= 50.00) %>%
  filter(Contamination <= 20.00) 

bins<-medium_bins$Bin_Id

write.table(bins, "lista_medium_bins", quote = F, row.names = F, col.names = F)
```

---
# Bins

```{r bi, eval=FALSE}
mkdir  -p 05.Bins/{Genoma,Proteoma}

cd 05.Bins

grep ">" *.fa

change_bin_name<-function(ruta, ambiente){
ruta_original<-getwd()
setwd(ruta)
filez <- list.files()
newname<-paste0(ambiente, "_", filez)
file.rename(from=filez, to=newname)
filez <- list.files()
file.rename(from=filez, to=sub(pattern="\\.", replacement="_", filez))
setwd(ruta_original)
}
```

---

# Phylotools

```{r phy, eval=FALSE}
library(phylotools)
library(tidyverse)

change_bin_name(home/user01/05.Bins, "pulque")

add_names_to_seqs <- function(nombre_del_archivo){
  filenames <- unlist(strsplit(nombre_del_archivo, "/"))
  filenames <- filenames[[grep("fa", filenames)]]
  divide <- unlist(strsplit(filenames, "\\."))
  bin_name <- divide[1]
  termination <- divide[2]
  old_name <- get.fasta.name(nombre_del_archivo)
  new_name <- paste0( bin_name, "-scaffold-", old_name) 
  ref2 <- data.frame(old_name, new_name)
  out_file <- paste0(bin_name, "_renamed", ".", termination)
  rename.fasta(infile = nombre_del_archivo, ref_table = ref2, outfile = out_file)
}

files <- list.files(".")
files <- paste0("/home/user01/05.Bins/pulque_Genoma/", files)
map(files, add_names_to_seqs)
files
```

---
# Renombrando Bins

```{r rename, eval=FALSE}
change_bin_name<-function(ruta){
ruta_original<-getwd()
setwd(ruta)
filez <- list.files()
file.rename(from=filez, to=sub(pattern="_renamed", replacement="", filez))
setwd(ruta_original)
}

change_bin_name("/home/user01/05.Bins/Genoma/01.Bins_named")
```

---
# Taxonomía

Para la taxonomía utilizamos GTDBTK

```{r gt, eval=FALSE}
mkdir 06.GTDBTK

```

Activamos el ambiente

```{r gt2, eval=FALSE}
conda activate gtdbtk-2.1.0
```

Exportamos base de datos

```{r gt3, eval=FALSE}
export GTDBTK_DATA_PATH=/home/programs/DB/release207_v2 
```


```{r gt4, eval=FALSE}
gtdbtk classify_wf --genome_dir /home/user01/05.Bins/pulque_Genoma/Bins_named --out_dir /home/megadriveCURSO/pulqueGTDBTK --cpus 4 -x fa 
```

---

# Taxonomía

```{r tax1, eval=FALSE}
GTDBK<-read.table("/home/megadriveCURSO/pulqueGTBDBTK/classify/gtbdbtk.bac120.summary.tsv", 
  sep = "\t", header = T, 
  na.strings ="", stringsAsFactors= F)%>%
  as_tibble()

pulque_gtdbtk<-GTDBK %>%
  select(user_genome, classification) %>%
  separate(classification, c("Domain", "Phylum", "Class", "Order",
                             "Family", "Genus", "Species"), sep= ";") %>%
  rename(Bin_name=user_genome)  %>%
  unite(Bin_name_2, c("Bin_name", "Phylum"), remove = FALSE) %>%
  select(Bin_name, Domain, Phylum, Class, Order, Family, Genus, 
         Species)
```

Haciendo metadatos

```{r meta10, eval=FALSE}
write.table(pulque_gtdbtk, file = "/home/megadriveCURSO/pulque/Metadatos.txt", sep="\t", quote = F,
            row.names = F, col.names = T)
```

---

IMAGEN

![]("path/")

```{r image1, eval=FALSE}
GTDBtk<-htn_gtdbtk %>%
  count(Domain, Phylum) %>%
  rename(Number_of_MAGs = n) %>%
  ggplot(aes(x = Domain, 
             y = Number_of_MAGs, fill = Phylum)) + 
  geom_bar(stat = "identity", position=position_dodge())+
  theme_minimal()
```


```{r imagen2, eval=FALSE}
library(plotly)
GTDBtk_p_fig <- ggplotly(GTDBtk)
```

---
 IMAGEN
 
 ![]("path/")
 
---

# Inferencia metabólica

```{r inf, eval=FALSE}
for i in /home/user01/05.Bins/pulque_Genoma/Bins_named/*.fa ; do prodigal -i $i -o $i.txt -a $i.faa ; done

grep ">" *.faa
```

KEGG

```{r kegg, eval=FALSE}
mkdir -p Kofamscan/{01.Proteomas,02.KO_results}
cd Kofamscan/
ln -s /home/user01/Bins/Proteoma/*.faa 01.Proteomas 

/home/programs/DB/kofam/kofam_scan-1.3.0/exec_annotation -o /home/user01/Kofamscan/htn_bins_63.fa.faa.txt /home/user01/Bins/Proteoma/htn_bins_63.fa.faa  --report-unannotated  --cpu 4

conda activate kofamscan
```

```{r kegg2, eval=FALSE}
htn_mapp<-read_ko("Kofamscan/02.KO_results/") %>%
    mapping_ko()
```

---

```{r kegg3, eval=FALSE}
Overview<-c("Central Metabolism", "Carbon Fixation", 
            "Nitrogen Metabolism", "Sulfur Metabolism", "Fermentation", 
            "Methane Metabolism")
Energy_metabolisms_pulque<-pulque_mapp %>%
  drop_na(Cycle) %>%
  get_subset_pathway(rbims_pathway, Overview) 
```

```{r kegg4, eval=FALSE}
plot_bubble(tibble_ko = Energy_metabolisms_htn,
            x_axis = Bin_name, 
            y_axis = Pathway_cycle,
            analysis="KEGG",
            calc="Percentage",
            range_size = c(1,10),
            y_labs=FALSE,
            x_labs=FALSE)  
```


---
IMAGEN

![]("path/")
---

# Graficando con metadatos

```{r kegg5, eval=FALSE}
Metadatos<-read_delim("/home/megadriveCURSO/pulque/Metadatos.txt", delim="\t")


plot_bubble(tibble_ko = Energy_metabolisms_htn,
            x_axis = Bin_name, 
            y_axis = Pathway_cycle,
            analysis="KEGG",
            data_experiment = Metadatos,
            calc="Percentage",
            color_character = Class,
            range_size = c(1,10),
            y_labs=FALSE,
            x_labs=FALSE) 
```

---

IMAGEN

![]("path/")
---

# Graficando una sola vía

```{r kegg6, eval=FALSE}
Fermentation_pulque<-htn_mapp %>%
  drop_na(Cycle) %>%
  get_subset_pathway(Cycle, "fermentation")
```

```{r}
#plot_heatmap(tibble_ko=Fermentation_pulque, 
#             y_axis=Genes,
#             analysis = "KEGG",
#             calc="Binary")
```

---

Imagen

![]("path/")

---

# Anotación

```{r anota, eval=FALSE}
conda activate Interpro

for i in $(ls *.faa); do sed -i "s/\*//g" $i; done

for i in $(ls *.faa); do interproscan.sh -cpu 4 -goterms -pa -i $i > Log_Interpro_Scan_$i.txt; done 


cat *.tsv > htn_interpro.tsv
```

```{r anota2, eval=FALSE}
library(rbims)
library(tidyverse)
```

---


```{r anota3, eval=FALSE}
interpro_Interpro_profile<-read_interpro(
  data_interpro = "09.Interpro/01.Proteomas/htn_interpro.tsv", 
  database="INTERPRO", profile = T) %>%
  filter(!str_detect(INTERPRO, "-"))

important_INTERPRO<-get_subset_pca(tibble_rbims=interpro_Interpro_profile, 
                                cos2_val=0.95,
                                analysis="INTERPRO")

plot_heatmap(important_INTERPRO, y_axis=INTERPRO, analysis = "INTERPRO", distance = T)

plot_heatmap(important_INTERPRO, y_axis=INTERPRO, analysis = "INTERPRO", distance = F)
```

