<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Proyecto 1: el microbioma del pulque</title>
    <meta charset="utf-8" />
    <meta name="author" content="Patricia Bustos, Montserrat García, Celso Cortés, Antonio Ovalle, Tobias Portillo, Eduardo Juscamayta, Armando Ocampo" />
    <meta name="date" content="2022-08-06" />
    <script src="libs/header-attrs-2.14/header-attrs.js"></script>
    <script src="libs/clipboard-2.0.6/clipboard.min.js"></script>
    <link href="libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.css" rel="stylesheet" />
    <script src="libs/xaringanExtra-clipboard-0.2.6/xaringanExtra-clipboard.js"></script>
    <script>window.xaringanExtraClipboard(null, {"button":"Copy Code","success":"Copied!","error":"Press Ctrl+C to Copy"})</script>
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Proyecto 1: el microbioma del pulque
]
.author[
### Patricia Bustos, Montserrat García, Celso Cortés, Antonio Ovalle, Tobias Portillo, Eduardo Juscamayta, Armando Ocampo
]
.institute[
### CDSD 2022
]
.date[
### 2022-08-06
]

---






# Flujo de trabajo

a) MaxBin

b) MetaBat

c) DASTool

d) Extracción de Bins

e) Taxonomía

f) Inferencia metabólica

g) Anotación

---

# MaxBin

Generación de enlaces simbólicos


```r
 mkdir -p 01.Metabat/{data,results}
 cd 01.Metabat/
 ln -s /home/diana/samples/pulque/data/pulque.fasta data/  
 ln -s /home/diana/samples/pulque/data/pulque_sorted.bam  data/ 
```

Activación de ambiente


```r
conda activate maxbin_env
```

Cálculo de bins


```r
run_MaxBin.pl -contig data/pulque.fasta -out results/maxbin -abund data/pulque-depth.txt -max_iteration 2
```

---

# MetaBat


```r
mkdir -p 02.Metabat/{data,results}
cd 02.Metabat/
ln -s /home/diana/samples/pulque/data/pulque.fasta data/  
ln -s /home/diana/samples/pulque/data/pulque_sorted.bam  data/ 
```

Activamos el ambiente

```r
conda activate metabat_env
```

Ensamble y archivos de profundidad


```r
jgi_summarize_bam_contig_depths --outputDepth data/pulque-depth.txt data/pulque_sorted.bam

metabat -i data/pulque.fasta -a data/pulque-depth.txt -o results/bins -t 4 --minCVSum 0 --saveCls -d -v --minCV 0.1 -m 2000
```

---
# DASTool

Refinamiento


Preparar datos de entrada


```r
Fasta_to_Scaffolds2Bin.sh -i /home/user01/01.Metabat -e fa &gt; htn.scaffolds2bin.tsv
```



```r
PATH=/home/programs:$PATH

cd /home/programs/DAS_Tool-1.1.2

R CMD INSTALL package/DASTool_1.1.2.tar.gz

/home/programs/DAS_Tool-1.1.2/DAS_Tool -i pulque_maxbin.contigs2bin.tsv,pulque_metabat.scaffolds2bin.tsv -l maxbin,metabat -c data/pulque.fasta -o results/pulque_bins --debug -t 4  --search_engine diamond --write_bins 1 
```


---

# Extracción de Bins


```r
mkdir 03.CheckM

cd 03.CheckM/
```

Activamos el ambiente


```r
conda activate checkm_env
```


```r
checkm  lineage_wf -t 4 -x fa /home/user01/04.DAS_tool/results/pulque_bins_DASTool_bins DAStools-log_htn  -f CheckM-DAS_Tool_bins.txt

sed -e '1,3d' CheckM-DAS_Tool_bins.txt | sed -e '37d' &gt;CheckM-DAS_Tool_bins_mod.txt
```

---

# Extracción de Bins


```r
library(tidyverse)


checkm&lt;-read.table("CheckM-DAS_Tool_bins_mod.txt", sep = "", header = F, na.strings ="", stringsAsFactors= F)


colnames(checkm)&lt;-c("Bin_Id", "Marker", "lineage", "Number_of_genomes", 
                         "Number_of_markers", "Number_of_marker_sets", 
                         "0", "1", "2", "3", "4", "5", "Completeness", 
                         "Contamination", "Strain_heterogeneity")  

good_bins&lt;-checkm %&gt;%
  select(Bin_Id, Marker, Completeness, Contamination) %&gt;%
  filter(Completeness &gt;= 50.00) %&gt;%
  filter(Contamination &lt;= 10.00) 
```

---

# Extracción de Bins


```r
medium_bins&lt;-checkm %&gt;%
  select(Bin_Id, Marker, Completeness, Contamination) %&gt;%
  filter(Completeness &gt;= 50.00) %&gt;%
  filter(Contamination &lt;= 20.00) 

bins&lt;-medium_bins$Bin_Id

write.table(bins, "lista_medium_bins", quote = F, row.names = F, col.names = F)
```

---
# Bins


```r
mkdir  -p 05.Bins/{Genoma,Proteoma}

cd 05.Bins

grep "&gt;" *.fa

change_bin_name&lt;-function(ruta, ambiente){
ruta_original&lt;-getwd()
setwd(ruta)
filez &lt;- list.files()
newname&lt;-paste0(ambiente, "_", filez)
file.rename(from=filez, to=newname)
filez &lt;- list.files()
file.rename(from=filez, to=sub(pattern="\\.", replacement="_", filez))
setwd(ruta_original)
}
```

---

# Phylotools


```r
library(phylotools)
library(tidyverse)

change_bin_name(home/user01/05.Bins, "pulque")

add_names_to_seqs &lt;- function(nombre_del_archivo){
  filenames &lt;- unlist(strsplit(nombre_del_archivo, "/"))
  filenames &lt;- filenames[[grep("fa", filenames)]]
  divide &lt;- unlist(strsplit(filenames, "\\."))
  bin_name &lt;- divide[1]
  termination &lt;- divide[2]
  old_name &lt;- get.fasta.name(nombre_del_archivo)
  new_name &lt;- paste0( bin_name, "-scaffold-", old_name) 
  ref2 &lt;- data.frame(old_name, new_name)
  out_file &lt;- paste0(bin_name, "_renamed", ".", termination)
  rename.fasta(infile = nombre_del_archivo, ref_table = ref2, outfile = out_file)
}

files &lt;- list.files(".")
files &lt;- paste0("/home/user01/05.Bins/pulque_Genoma/", files)
map(files, add_names_to_seqs)
files
```

---
# Renombrando Bins


```r
change_bin_name&lt;-function(ruta){
ruta_original&lt;-getwd()
setwd(ruta)
filez &lt;- list.files()
file.rename(from=filez, to=sub(pattern="_renamed", replacement="", filez))
setwd(ruta_original)
}

change_bin_name("/home/user01/05.Bins/Genoma/01.Bins_named")
```

---
# Taxonomía

Para la taxonomía utilizamos GTDBTK


```r
mkdir 06.GTDBTK
```

Activamos el ambiente


```r
conda activate gtdbtk-2.1.0
```

Exportamos base de datos


```r
export GTDBTK_DATA_PATH=/home/programs/DB/release207_v2 
```



```r
gtdbtk classify_wf --genome_dir /home/user01/05.Bins/pulque_Genoma/Bins_named --out_dir /home/megadriveCURSO/pulqueGTDBTK --cpus 4 -x fa 
```

---

# Taxonomía


```r
GTDBK&lt;-read.table("/home/megadriveCURSO/pulqueGTBDBTK/classify/gtbdbtk.bac120.summary.tsv", 
  sep = "\t", header = T, 
  na.strings ="", stringsAsFactors= F)%&gt;%
  as_tibble()

pulque_gtdbtk&lt;-GTDBK %&gt;%
  select(user_genome, classification) %&gt;%
  separate(classification, c("Domain", "Phylum", "Class", "Order",
                             "Family", "Genus", "Species"), sep= ";") %&gt;%
  rename(Bin_name=user_genome)  %&gt;%
  unite(Bin_name_2, c("Bin_name", "Phylum"), remove = FALSE) %&gt;%
  select(Bin_name, Domain, Phylum, Class, Order, Family, Genus, 
         Species)
```

Haciendo metadatos


```r
write.table(pulque_gtdbtk, file = "/home/megadriveCURSO/pulque/Metadatos.txt", sep="\t", quote = F,
            row.names = F, col.names = T)
```

---
#### Taxonomía

![](presentacion_proyecto_files/imagenes/domain.png)

---



```r
GTDBtk&lt;-htn_gtdbtk %&gt;%
  count(Domain, Phylum) %&gt;%
  rename(Number_of_MAGs = n) %&gt;%
  ggplot(aes(x = Domain, 
             y = Number_of_MAGs, fill = Phylum)) + 
  geom_bar(stat = "identity", position=position_dodge())+
  theme_minimal()
```



```r
library(plotly)
GTDBtk_p_fig &lt;- ggplotly(GTDBtk)
```

---
#### Dominio Proteico

![](presentacion_proyecto_files/imagenes/protein.png)
 
---

# Inferencia metabólica


```r
for i in /home/user01/05.Bins/pulque_Genoma/Bins_named/*.fa ; do prodigal -i $i -o $i.txt -a $i.faa ; done

grep "&gt;" *.faa
```

KEGG


```r
mkdir -p Kofamscan/{01.Proteomas,02.KO_results}
cd Kofamscan/
ln -s /home/user01/Bins/Proteoma/*.faa 01.Proteomas 

/home/programs/DB/kofam/kofam_scan-1.3.0/exec_annotation -o /home/user01/Kofamscan/htn_bins_63.fa.faa.txt /home/user01/Bins/Proteoma/htn_bins_63.fa.faa  --report-unannotated  --cpu 4

conda activate kofamscan
```


```r
htn_mapp&lt;-read_ko("Kofamscan/02.KO_results/") %&gt;%
    mapping_ko()
```

---


```r
Overview&lt;-c("Central Metabolism", "Carbon Fixation", 
            "Nitrogen Metabolism", "Sulfur Metabolism", "Fermentation", 
            "Methane Metabolism")
Energy_metabolisms_pulque&lt;-pulque_mapp %&gt;%
  drop_na(Cycle) %&gt;%
  get_subset_pathway(rbims_pathway, Overview) 
```


```r
plot_bubble(tibble_ko = Energy_metabolisms_htn,
            x_axis = Bin_name, 
            y_axis = Pathway_cycle,
            analysis="KEGG",
            calc="Percentage",
            range_size = c(1,10),
            y_labs=FALSE,
            x_labs=FALSE)  
```


---
#### Vías metabólicas

![](presentacion_proyecto_files/imagenes/buble_plot.png)
---

# Graficando con metadatos


```r
Metadatos&lt;-read_delim("/home/megadriveCURSO/pulque/Metadatos.txt", delim="\t")


plot_bubble(tibble_ko = Energy_metabolisms_htn,
            x_axis = Bin_name, 
            y_axis = Pathway_cycle,
            analysis="KEGG",
            data_experiment = Metadatos,
            calc="Percentage",
            color_character = Class,
            range_size = c(1,10),
            y_labs=FALSE,
            x_labs=FALSE) 
```


---

# Graficando una sola vía


```r
Fermentation_pulque&lt;-htn_mapp %&gt;%
  drop_na(Cycle) %&gt;%
  get_subset_pathway(Cycle, "fermentation")
```


```r
#plot_heatmap(tibble_ko=Fermentation_pulque, 
#             y_axis=Genes,
#             analysis = "KEGG",
#             calc="Binary")
```

---

#### Vía de fermentación

![](presentacion_proyecto_files/imagenes/fermentation_pathway.png)

---

# Anotación


```r
conda activate Interpro

for i in $(ls *.faa); do sed -i "s/\*//g" $i; done

for i in $(ls *.faa); do interproscan.sh -cpu 4 -goterms -pa -i $i &gt; Log_Interpro_Scan_$i.txt; done 


cat *.tsv &gt; htn_interpro.tsv
```


```r
library(rbims)
library(tidyverse)
```

---



```r
interpro_Interpro_profile&lt;-read_interpro(
  data_interpro = "home/megadriveCURSO/plque/05.Bins/pulque_Genoma/bins_named/pulque_interpro.tsv", 
  database="INTERPRO", profile = T) %&gt;%
  filter(!str_detect(INTERPRO, "-"))

important_INTERPRO&lt;-get_subset_pca(tibble_rbims=interpro_Interpro_profile, 
                                cos2_val=0.97,
                                analysis="INTERPRO")

plot_heatmap(important_INTERPRO, y_axis=INTERPRO, analysis = "INTERPRO", distance = T)

plot_heatmap(important_INTERPRO, y_axis=INTERPRO, analysis = "INTERPRO", distance = F)
```

---

    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
